### ISSUE TITLE
Kubernetes 1.6 may use etcd V3 by default, migration may break Calico

### ISSUE THREAD

**Issue by fasaxc** [projectcalico/cni-plugin#281]
Came up in a discussion with **mattymo** in calicousers.  Lots of users share their etcd between k8s and Calico.  In k8s 1.6, k8s is moving to use the etcdv3 API by default.  

Although etcd v3 still supports the v2 API, which we use, the worry is that the migration of k8s data from v2 to v3 will break Calico.

**Comment by fasaxc** [comment]
Marking as a shipstopper for v2.1.0 because it could be a train wreck if uses start updating and find that Calico is broken.  OTOH, might be less serious once we get more of a handle on it!

**Comment by tmjd** [comment]
Here is some info I found while looking into this.
Here is the script that I think is intended to be used for migration https://github.com/kubernetes/kubernetes/blob/master/cluster/images/etcd/migrate-if-needed.sh
It currently doesn't do any kind of selective migration and looking at the options for the `etcdctl migrate` command, it does not look like it is even an option. https://github.com/coreos/etcd/blob/master/etcdctl/README.md#migrate-options 

Since it wasn't clear in the migrate docs I asked in the #etcd channel on freenode and was told that the migrate command won't touch the v2 keys when migrating from v2 to v3.  So this may not be a concern at all then.

Here is the k8s Issue for tracking docs for the upgrade 
https://github.com/kubernetes/kubernetes.github.io/issues/2172

Here is a CoreOS blog entry about etcd migration for k8s.
https://coreos.com/blog/migrating-applications-etcd-v3.html

**Comment by tmjd** [comment]
More confirmation that migration won't wipe out the v2 data.
https://github.com/coreos/etcd/issues/7406

**Comment by caseydavenport** [comment]
Downgraded given @tmjd says this won't be an issue.

Feel free to close or repurpose this issue.  We may want documentation for this (a line in the k8s upgrade docs?)

**Comment by timothysc** [comment]
You can run 1 etcd for calico and one for the cluster backend as an easy exit.

**Comment by tmjd** [comment]
Here is the Upgrade 1.6 docs in k8s https://kubernetes.io/docs/admin/upgrade-1-6/ add by https://github.com/kubernetes/kubernetes.github.io/pull/2763

tl;dr It is possible to use 1.6 with etcdv2 by setting a flag but for the long term, etcd should be upgraded.
